### PR TITLE
Preventing first turn sound notification if this turn was noticed after word then.

### PR DESCRIPTION
**Pull request by bykoianko**
В при движении по маршруту, если два последующих поворота идут близко, то пользователь услышит: Поворот на право, затем поворот на лево. В данном случае не должно быть оповещение Поворот на лево через 200 метров после прохождения поворота на право.

Данный PR решает эту задачу.

https://trello.com/c/3sgwHYi1/1883--

### CODE REVIEW

**Comment by mapsmetest** [comment]
Jenkins asks: can anyone have a look at this request?
If you want Jenkins to compile and test this, write "test this please" or JT_TP (without the underscore in the middle).

**Comment by gardster** [comment]
LGTM

File: routing/turns_sound.cpp [hunk 1]
```diff
@@ -73,6 +73,10 @@ void TurnsSound::GenerateTurnSound(vector<TurnItemDist> const & turns, vector<st
   if (secondNotification.empty())
     return;
   turnNotifications.emplace_back(move(secondNotification));
+  // Turn notification with word "Then" (about the second turn) was pronounced.
```
Comment (kshalnev) [inline routing/turns_sound.cpp]: As was discussed: 'will pronounced'. please fix comments in the code.

File: routing/turns_sound.cpp [hunk 2]
```diff
@@ -122,6 +134,7 @@ string TurnsSound::GenerateFirstTurnSound(TurnItem const & turn, double distance
     return GenerateTurnText(0 /* distanceUnits */, turn.m_exitNum,
                             false /* useThenInsteadOfDistance */,
                             turn.m_turn, m_settings.GetLengthUnits());
+    FastForwardFirstTurnNotification();
```
Comment (kshalnev) [inline routing/turns_sound.cpp]: call function after return!

File: routing/turns_sound.hpp [hunk 3]
```diff
@@ -56,6 +56,11 @@ class TurnsSound
   /// After the second notification has been pronounced
   /// m_nextTurnNotificationProgress == Second.
   PronouncedNotification m_nextTurnNotificationProgress;
+  /// The flag is set to true if notification about the second turn was pronounced.
```
Comment (kshalnev) [inline routing/turns_sound.hpp]: By the way, according to coding standard private section follows after public and protected.

File: routing/turns_sound.cpp (hunk 1 above)
Comment (bykoianko) [inline routing/turns_sound.cpp]: ok. Will be pronounced.

File: routing/turns_sound.cpp (hunk 2 above)
Comment (bykoianko) [inline routing/turns_sound.cpp]: > call function after return!

Fixed

> unit tests?

Yes, I'll write it.

File: routing/turns_sound.hpp (hunk 3 above)
Comment (bykoianko) [inline routing/turns_sound.hpp]: Which standard are you talking about?
As far as I know there's no restriction for the order in our coding standard.
We've discussed this topic recently.

**Comment by bykoianko** [comment]
@kshalnev PTAL

**Comment by kshalnev** [comment]
LGTM

**Pull request merged**